### PR TITLE
Adding backup workflow

### PR DESCRIPTION
**Pull request by shaheislamdfe**
### Context

- Ticket: https://trello.com/c/drLgfoCB/2114-add-dr-workflows-to-cpd

### Changes proposed in this pull request

Adding DR workflows

### Guidance to review

Testing: 

https://github.com/DFE-Digital/early-careers-framework/actions/runs/11817678838/job/32923589830
https://github.com/DFE-Digital/early-careers-framework/actions/runs/11820438594/job/32932736398

### CODE REVIEW

File: .github/workflows/backup_nightly_database.yml [hunk 1]
```diff
@@ -1,18 +1,87 @@
...
+        required: false
+        type: string
+        default: default
+      db-server:
+        description: |
+          Name of the database server. Default is the live server. When backing up a point-in-time (PTR) server, use the full name of the PTR server. (Optional)
+
   schedule:
-    - cron: "0 3 * * *" # 03:00 UTC
+    - cron: "0 4 * * *" # 04:00 UTC
```
Comment (saliceti) [inline .github/workflows/backup_nightly_database.yml:25]: The team is used to this time, can we keep it?

File: .github/workflows/restore_snapshot_database.yml [hunk 2]
```diff
@@ -1,25 +1,72 @@
-name: Restore Snapshot DB from production DB
```
Comment (cwrw) [inline .github/workflows/restore_snapshot_database.yml:1]: could we avoid touching the snapshot please? this is a different workflow for a test db we don't need to change

File: .github/workflows/backup_nightly_database.yml (hunk 1 above)
Comment (cwrw) [reply .github/workflows/backup_nightly_database.yml:26]: yes please can we not change that

**Comment by shaheislamdfe** [comment]
[Add DR workflows to CPD](https://trello.com/c/drLgfoCB/2114-add-dr-workflows-to-cpd)

**Review by cwrw — COMMENTED** [review]
looks good past the one comment

**Review by saliceti — APPROVED** [review]

**Pull request merged**